extension/app/ts/simulation/simulator.ts

```diff
@@ -54,7 +54,7 @@ export class Simulator {
 		for (let i = 0; i < transactions.length; i++) {
 			resultPromises.push(this.visualizeTransaction(transactions[i], blockNumber, multicallResults[i]))
 		}
-		return Promise.all(resultPromises)
+		return await Promise.all(resultPromises)
```

Comment:
Technically, I believe these are the same thing.  If you return a Promise from an async function, it "does the right thing" and adds the returned promise as a continuation.
```ts
async function apple() { return new Promise(resolve => setTimeout(() => resolve(1), 1000)) } 
await apple()
// <1 second later>
// <- 1
```

Reply:
yeah, I think they are the exactly the same. It somehow feels right to add Await there 🤔

Reply:
I agree, the `await` is good as you should generally always await promises.  Just commenting that it isn't strictly necessary.